app/frontend/chat_interface.py

```diff
@@ -224,6 +224,120 @@ def resend_confirmation(email: str) -> dict | None:
     return call_auth_endpoint('/resend-confirmation', {'email': email}, method='POST')
 
 
+def upload_image_to_backend(uploaded_file) -> dict | None:
+    """Upload and validate an image using the FastAPI backend.
+    
+    Args:
+        uploaded_file: Streamlit UploadedFile object
+        
+    Returns:
+        dict: Image metadata if successful, None if failed
+    """
+    try:
+        url = f'{API_BASE_URL}/api/v1/images/upload'
+        
+        # Reset file pointer to beginning
+        uploaded_file.seek(0)
+        
+        # Prepare multipart form data
+        files = {'file': (uploaded_file.name, uploaded_file, uploaded_file.type)}
+        
+        response = requests.post(url, files=files, timeout=settings.api_timeout)
+        
+        if response.status_code == 200:
+            return response.json()
+        elif response.status_code == 400:
+            # Validation error
+            error_detail = response.json().get('detail', 'Image validation failed.')
+            st.error(f'Image validation failed: {error_detail}')
+            return None
+        else:
+            st.error('Failed to upload image. Please try again.')
+            if STREAMLIT_ENV == 'dev':
+                st.error(f'API Error: {response.status_code} - {response.text}')
+            return None
+            
+    except requests.exceptions.RequestException as e:
+        st.error('Connection Error: Unable to connect to the API server.')
+        if STREAMLIT_ENV == 'dev':
+            st.error(f'Debug info: {str(e)}')
+        return None
+
+
+def validate_image_client(uploaded_file) -> bool:
+    """Quick client-side checks before sending to backend.
+
+    Performs shallow validation to provide fast feedback and avoid
+    unnecessary network calls for obviously invalid files.
+
+    Checks:
+    - Extension in ['jpg','jpeg','png','webp']
+    - Size <= 3.75 MB (to match backend limit)
+    - Resolution <= 4096 x 4096
+
+    Returns:
+        True if all checks pass, otherwise shows a UI message and returns False.
+    """
+    try:
+        allowed_exts = {'jpg', 'jpeg', 'png', 'webp'}
+
+        filename = getattr(uploaded_file, 'name', '') or ''
+        ext = filename.rsplit('.', 1)[-1].lower() if '.' in filename else ''
+        if ext not in allowed_exts:
+            st.warning('Unsupported image file type. Allowed: jpg, jpeg, png, webp.')
+            return False
+
+        file_size = getattr(uploaded_file, 'size', None)
+        if file_size is None:
+            try:
+                file_size = len(uploaded_file.getbuffer())  # type: ignore[attr-defined]
+            except Exception:
+                file_size = None
+
+        if file_size is not None:
+            max_bytes = int(3.75 * 1024 * 1024)
+            if file_size > max_bytes:
+                st.warning('Image is too large. Maximum allowed size is 3.75 MB.')
+                return False
```

Comment:
Hardcoded size limit (3.75 MB) differs from backend limit (5 MB). Synchronize the client-side size check with backend configuration.
```
max_bytes = int(settings.image_max_size_mb * 1024 * 1024)
if file_size > max_bytes:
    st.warning(f'Image is too large. Maximum allowed size is {settings.image_max_size_mb} MB.')
    return False
```